crypto/rsa/rsa_gen.c

```diff
@@ -71,6 +71,173 @@ int RSA_generate_multi_prime_key(RSA *rsa, int bits, int primes,
     return rsa_keygen(rsa->libctx, rsa, bits, primes, e_value, cb, 0);
 }
 
+DEFINE_STACK_OF(BIGNUM)
+
+/*
+ * Given input values, q, p, n, d and e, derive the exponents
+ * and coefficients for each prime in this key, placing the result
+ * on their respective exps and coeffs stacks
+ */
+int ossl_rsa_multiprime_derive(RSA *rsa, int bits, int primes,
```

Comment:
This should be in`#ifndef FIPS_MODULE` shouldn't it? Also would it be actually possible to refactor the keygen below to call this after it generates the factors?

Reply:
sure